mesonbuild/dependencies/python.py

```diff
@@ -271,6 +271,18 @@ def get_windows_link_args(self) -> T.Optional[T.List[str]]:
                         libpath = Path(f'python{vernum}.dll')
                 else:
                     libpath = Path('libs') / f'python{vernum}.lib'
+                    # For a debug build, pyconfig.h forces linking with
+                    # pythonX_d.lib (see meson#10776). This cannot be avoided
+                    # and won't work unless we also have a debug build of
+                    # Python itself. So error out here.
+                    buildtype = self.env.coredata.get_option(mesonlib.OptionKey('buildtype'))
+                    assert isinstance(buildtype, str)
+                    debug = self.env.coredata.get_option(mesonlib.OptionKey('debug'))
+                    is_debug_build = debug or buildtype.startswith('debug')
+                    if is_debug_build and not self.variables.get('Py_DEBUG'):
+                        msg = "Using a debug build type with MSVC or an MSVC-compatible compiler is " \
+                              "only supported if the Python interpreter is also a debug build."
```

Comment:
Instead of merely saying it's "unsupported" I think we should actively state something to the effect of, unless you go out of your way to hack the header and make it work it will almost certainly result in a failing build.

Reply:
Unsupported sounds like "you can do it but don't report bugs to us".

"Results in a failing build" sounds like "if your build fails, this is the root cause, so you can stop trying to further debug it".

Reply:
Makes sense. Like this?
```python
msg = "Using a debug build type with MSVC or an MSVC-compatible compiler when " \
      "the Python interpreter is not also a debug build will almost certainly. " \
      "result in a failed build. Prefer using a release build type or a debug "\
      "Python interpreter."
```